eZ/Publish/Core/Repository/Tests/Service/Mock/ContentTest.php

```diff
@@ -3170,6 +3173,109 @@ public function testUpdateContentThrowsUnauthorizedException()
         $mockedService->updateContent($versionInfo, $contentUpdateStruct);
     }
 
+    /**
+     * @covers \eZ\Publish\Core\Repository\ContentService::updateContent
+     */
+    public function testUpdateContentGetsProperFieldsToUpdate(): void
```

Comment:
This test doesn't bring enough value. What we need here is an integration test. Looking at the issue steps to reproduce writing such test should be straightforward.

Reply:
I'm really not sure why would we want to test an edge case. The real problem is that `getFieldsForUpdate` is invalid. The fact that we have an exception is just an edge case resulting from the original problem which is why fields to update are unnecessarily added. Isn't this rather a problem with the `ContentMapper`?

Reply:
Please, note that Fieldgroup limitation is outside of `ezplatform-kernel` package. How would an integration test apply to kernel then if we are doing a change to the kernel itself?

Reply:
I would expect that package I'm updating has a representation in a unit test in the exact same package.

Reply:
>I'm really not sure why would we want to test an edge case. 

Outside of the PR, but one of the reasons is that to not introduce bug again when code is touched to fix another edge case or refactored.

Reply:
That's why I've actually tested it in `Mock/ContentTest`. As mentioned below `FieldGroupLimitation` being outside of this package is irrelevant.

Reply:
@barw4 ok, after more careful investigation I somewhat have an idea about what is happening here, but not fully. I agree that due to `FieldGroupLimitation` being part of Ibexa DXP, outside of the core, it's not possible to cover it with integration-level testing.

What seems to be the bug is that `ContentMapper::getFieldsForUpdate` returns incorrect set of fields for update, right?
I also see why it happens, however I wasn't able to fully reproduce this on API due to issue being more complex than I initially thought.

If `ContentMapper` is broken, you actually need to cover `ContentMapper`, not `ContentService::updateContent`.

_**I need to see an unit test which shows what result is expected from `ContentMapper::getFieldsForUpdate`**_. Please don't do partial mocks of classes under tests, this is anti-pattern we didn't have time to clean up.

Reply:
> What seems to be the bug is that ContentMapper::getFieldsForUpdate returns incorrect set of fields for update, right?

@alongosz Yes, this is exactly the case. Afaik we have no dedicated `ContentMapper` test anywhere, so I'll create one instead of leave `Mock\ContentTest` intact.

Reply:
> > What seems to be the bug is that ContentMapper::getFieldsForUpdate returns incorrect set of fields for update, right?
> 
> @alongosz Yes, this is exactly the case. Afaik we have no dedicated `ContentMapper` test anywhere, so I'll create one instead and leave `Mock\ContentTest` intact.

Hint: `\Ibexa\Tests\Core\Repository\Mapper\ContentMapperTest` in `tests/lib/Repository/Mapper` is a good choice for that test class, to keep -> ibexa upgrade path smooth.

Reply:
@alongosz resolved in https://github.com/ezsystems/ezplatform-kernel/pull/368/commits/2c8c2f58b3329cb52005e3a5ec7d9726d411d201